python/semantic_kernel/connectors/memory/azure_cosmos_db/azure_cosmos_db_no_sql_collection.py

```diff
@@ -0,0 +1,212 @@
+
+from azure.cosmos.aio import CosmosClient
+from azure.cosmos.exceptions import CosmosResourceNotFoundError
+from azure.cosmos.partition_key import PartitionKey
+
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_base import AzureCosmosDBNoSQLBase
+from semantic_kernel.connectors.memory.azure_cosmos_db.azure_cosmos_db_no_sql_composite_key import (
+    AzureCosmosDBNoSQLCompositeKey,
+)
+from semantic_kernel.connectors.memory.azure_cosmos_db.utils import (
+    COSMOS_ITEM_ID_PROPERTY_NAME,
+    build_query_parameters,
+    create_default_indexing_policy,
+    create_default_vector_embedding_policy,
+    get_key,
+    get_partition_key,
+)
+from semantic_kernel.data.record_definition.vector_store_model_definition import VectorStoreRecordDefinition
+from semantic_kernel.data.vector_storage.vector_store_record_collection import VectorStoreRecordCollection
+from semantic_kernel.exceptions.memory_connector_exceptions import (
+    MemoryConnectorException,
+    MemoryConnectorResourceNotFound,
+    VectorStoreModelDeserializationException,
+)
+from semantic_kernel.kernel_types import OneOrMany
+from semantic_kernel.utils.experimental_decorator import experimental_class
+
+TModel = TypeVar("TModel")
+TKey = TypeVar("TKey", str, AzureCosmosDBNoSQLCompositeKey)
+
+
+@experimental_class
+class AzureCosmosDBNoSQLCollection(AzureCosmosDBNoSQLBase, VectorStoreRecordCollection[TKey, TModel]):
+    """An Azure Cosmos DB NoSQL collection stores documents in a Azure Cosmos DB NoSQL account."""
+
+    partition_key: PartitionKey
+
+    def __init__(
+        self,
+        data_model_type: type[TModel],
+        database_name: str,
+        collection_name: str,
+        data_model_definition: VectorStoreRecordDefinition | None = None,
+        url: str | None = None,
+        key: str | None = None,
+        cosmos_client: CosmosClient | None = None,
+        partition_key: PartitionKey | None = None,
+        create_database: bool = False,
+    ):
+        """Initializes a new instance of the AzureCosmosDBNoSQLCollection class.
+
+        Args:
+            data_model_type (type[TModel]): The type of the data model.
+            database_name (str): The name of the database. Used to create a database proxy if not provided.
+            collection_name (str): The name of the collection.
+            data_model_definition (VectorStoreRecordDefinition): The definition of the data model. Defaults to None.
+            url (str): The URL of the Azure Cosmos DB NoSQL account. Defaults to None.
+            key (str): The key of the Azure Cosmos DB NoSQL account. Defaults to None.
+            cosmos_client (CosmosClient): The custom Azure Cosmos DB NoSQL client whose lifetime is managed by the user.
+            partition_key (PartitionKey): The partition key. Defaults to None. If not provided, the partition
+                                          key will be based on the key field of the data model definition.
```

Comment:
would be good to add this link here: https://learn.microsoft.com/en-us/azure/cosmos-db/partitioning-overview partition key choice is so crucial for a good experience with Cosmos that we should make sure people consider it and not use the default always (although it is a good choice for many cases), especially the note that aligning your Partition Key to commonly used filters is a useful when in the context of search!